provider/pkg/provider/container.go

```diff
@@ -698,3 +702,117 @@ func updateNodePoolConfig(diffPropPath, targetPropPath resource.PropertyPath,
 		return nil
 	}
 }
+
+// *** Support for method invocations on container resources ***
+// Most of this was adapted from a pattern established in pulumi-yaml:
+// https://github.com/pulumi/pulumi-yaml/blob/a36000e5bf6b64c050ec8777578aea3630dc1b7f/pkg/pulumiyaml/run.go
+
+// callableResource models the receiver resource for a method call in generic terms.
+// While currently we only have the getKubeconfig method on the cluster resource,
+// this can be reused for others in the future and should move to a separate package/file.
+// This bag-based output shape defined here helps us avoid adding an unholy dependency on the
+// generated Go SDK here (albeit which might have given us strongly typed outputs instead).
+type callableResource interface {
+	GetOutput(k string) pulumi.Output
+	CustomResource() *pulumi.CustomResourceState
+	ProviderResource() *pulumi.ProviderResourceState
+}
+
+type customResourceState struct {
+	pulumi.CustomResourceState
+	name    string
+	Outputs pulumi.MapOutput `pulumi:""`
```

Comment:
Is the `""` tag correct?

Reply:
I have to admit I copied this from the yaml implementation I referenced. It does seem to capture the outputs as expected - though why it works might not be obvious.